### DISCUSSION TITLE
Mend UI Silent Mode - Configuration Ignored

### DISCUSSION THREAD

**Discussion by tcd156** [renovatebot/renovate#30602] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

Running the Mend Renovate Github app in silent mode uses our global configuration repository and appears to do exactly what we want.

For example, grouping Terraform modules into a single PR works *while that repository is configured with Silent mode to be enabled*.

As soon as I disable Silent mode on that repository, it completely reverts to the Renovate default configuration and no longer uses my global configuration repo.

Our global repo is in our orgs top level, it's name is `renovate-config`, and we have a `default.json` file with the configuration.

As I said though, it works when the repository is configured for Silent mode. Disabling Silent mode then stops using this configuration.

What am I missing?

`renovate-config/default.json`:
```
{
  "extends": [
      "config:best-practices"
  ],
  "autodiscover": true,
  "requireConfig": "optional",
  "branchPrefix": "renovate/",
  "platform": "github",
  "updateInternalDeps": true,
  "labels": [
    "dependencies",
    "renovatebot"
  ],
  "hostRules": [
    {
      "matchHost": "spacelift.io",
      "token": "{{ secrets.SPACELIFT_TOKEN }}"
    }
  ],
  "packageRules": [
    {
      "matchDatasources": ["terraform-module"],
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "groupName": "Terraform modules (non-major)"
    },
    {
      "matchDatasources": ["terraform-module"],
      "matchUpdateTypes": ["major"],
      "groupName": "Terraform modules (major)"
    },
    {
      "matchDatasources": ["terraform-provider"],
      "matchPackageNames": ["hashicorp/terraform"],
      "allowedVersions": "<=1.5.7"
    },
    {
      "matchDatasources": ["terraform-provider"],
      "matchUpdateTypes": ["minor", "patch", "pin", "digest"],
      "groupName": "Terraform providers (non-major)"
    },
    {
      "matchDatasources": ["terraform-provider"],
      "matchUpdateTypes": ["major"],
      "groupName": "Terraform providers (major)"
    }
  ],
  "vulnerabilityAlerts": {
    "labels": [
      "dependencies",
      "renovatebot",
      "security"
    ]
  }
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Please post both job logs to gists and link here

**Comment by tcd156** [comment]
Org silent mode on - repo Silent mode ON:
https://gist.github.com/tcd156/b3d59615f8660aac4b394e748b2693e8

Org silent mode on - repo Silent mode OFF:
https://gist.github.com/tcd156/e9d1278e51ae6a8eb242b3934f6df2f2

I notice it seems that maybe some caching is coming into play. I have a Spacelift token and hostRules to configure it. It works but when I tested changing the rules to verify that my global config was working or not, it seemed it took some time for those rules to be picked up.

I'm trying to understand if self-hosting would resolve these issues, because so far, I haven't had a lot of luck using this configuration with the Github app.

**Reply by rarkins** [reply]
`DEBUG: No renovate config file found`

You have no repo config, which is where you extend from global configs. It does not get extended automatically. 

When in silent mode it assumes you'll onboard the repo with default settings, including extending your global config.

**Reply by tcd156** [reply]
OK, sorry I must be confused - I thought that having the `renovate-config` repo, Renovate would read from the `default.json` file  and use those settings as the global config.

I definitely understand about the inheritance. I thought that config would always be used.

So are you saying that's not the case? Do I have to have a file in each repo regardless to inherit to handle that case?

Apologies for the confusion.

**Reply by rarkins** [reply]
If you have a renovate-config repo then it will be proposed in the onboarding config.  If you use silent mode then the onboarding config will be assumed. But if you have an empty repo config then the renovate-config won't be implicitly included. 

What you're after sounds like the inheritConfig, which is implicit. Unfortunately it's not supported in the app yet, but soon.

Cc @nabeelsaabna

**Reply by tcd156** [reply]
Understood! Thank's for the clarification!

Sounds like I could:
1. Self-host Renovate to get this functionality.
2. Create a `renovate.json` configuration in each repository that includes a pointer to the configuration repo.

I'm essentially attempting to onboard repos if I can to replace Dependabot. Our repos don't have any custom Dependabot configurations. I thought I could mimic that configuration not to have to include this file in each repository. 

Onboarding PRs are obviously not bad, and probably is the way to go anyway, but that's where my question was stemming from.

Is my thinking correct?

**Reply by nabeelsaabna** [reply]
@rarkins we prioritized automatic support for `inheritConfig` in the hosted app

**Comment by jamietanna** [comment, marked as answer]
Since the last updates, `inheritConfig` is now available on the Mend Developer Platform - if you see any issues with it, please raise a new Discussion to let us know!
